Neos.Neos/Classes/FrontendRouting/Projection/DocumentUriPathProjection.php

```diff
@@ -445,6 +445,21 @@ private function copyVariants(
                 ->withOriginDimensionSpacePoint($targetOrigin)
                 ->withoutSiblings();
 
+            // check the parent in the "target" dimensionSpacePoint for the "URI prefix",
+            // may be different, see neos/neos-development-collection#5090
+            $parentNode = $this->tryGetNode(fn () => $this->getState()->getByIdAndDimensionSpacePointHash(
+                $sourceNode->getParentNodeAggregateId(),
+                $interdimensionalSibling->dimensionSpacePoint->hash
+            ));
+            if ($parentNode !== null) {
+                $uriPathSegment = basename($sourceNode->getUriPath());
+                $uriPath = $parentNode->getUriPath() === ''
+                    ? $uriPathSegment
+                    : $parentNode->getUriPath() . '/' . $uriPathSegment;
```

Comment:
I'm not sure how we "calculate" the uri path in other places, but I don't remember having used `basename()` (not that I dislike it, but IMO it should be unified – potentially centralized to some `DocumentNodeInfo::withUriPathSegment(string $newSegment)`!?

Reply:
Well, `withUriPathSegment(…)` does not help here, the `basename(…)` is used to get the segment I would need to pass here.

But I will get rid of basename and use code like in `whenNodePropertiesWereSet(…)` later in the class.